contracts/modules/VersionManager.sol

```diff
@@ -188,16 +202,17 @@ contract VersionManager is IVersionManager, IModule, BaseFeature, Owned {
      * @inheritdoc IFeature
      */
     function init(address _wallet) public override(IModule, BaseFeature) onlyWallet(_wallet) {
-        doUpgradeWallet(_wallet);
+        doUpgradeWallet(_wallet, newWalletVersion);
     }
 
     /**
-     * @notice Upgrade a wallet to the latest version.
+     * @notice Upgrade a wallet to a new version.
      * @dev It's cheaper to pass features to init as calldata than reading them from storage
-     * @param _wallet the wallet to upgrrade
+     * @param _wallet the wallet to upgrade
+     * @param _version the new version
      */
-    function upgradeWallet(address _wallet) external onlyWalletOwnerOrFeature(_wallet) onlyWhenUnlocked(_wallet) {
-        doUpgradeWallet(_wallet);
+    function upgradeWallet(address _wallet, uint256 _version) external onlyWalletOwnerOrFeature(_wallet) onlyWhenUnlocked(_wallet) {
```

Comment:
What is the reason to let the user choose the target version? Isn't there now the risk that a client upgrades to a version that contains a bug and we cannot prevent it?

Reply:
Yes, that's the downside and the reason why I had prevented it to begin with. But the clients requested this in case the client is out of sync with the latest version (e.g. user wallet is on V5, client only supports V7 and on chain the last version is V10 => need to update user to V7). I am hesitating to add a way for the multisig to blacklist a past version but not sure I want to overcomplicate the VersionManager.

Reply:
Agree on not complicating the VersionManager.